comp/api/api/apiimpl/internal/agent/agent_test.go

```diff
@@ -0,0 +1,173 @@
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+	"github.com/DataDog/datadog-agent/pkg/version"
+
+	// third-party dependencies
+	"github.com/gorilla/mux"
+	"go.uber.org/fx"
+)
+
+type handlerdeps struct {
+	fx.In
+
+	FlareComp             flare.Component
+	Server                dogstatsdServer.Component
+	ServerDebug           dogstatsddebug.Component
+	Wmeta                 workloadmeta.Component
+	LogsAgent             optional.Option[logsAgent.Component]
+	HostMetadata          host.Component
+	InvAgent              inventoryagent.Component
+	Demux                 demultiplexer.Component
+	InvHost               inventoryhost.Component
+	SecretResolver        secrets.Component
+	InvChecks             inventorychecks.Component
+	PkgSigning            packagesigning.Component
+	StatusComponent       status.Component
+	Collector             optional.Option[collector.Component]
+	EventPlatformReceiver eventplatformreceiver.Component
+	Ac                    autodiscovery.Mock
+}
+
+func getComponentDeps(t *testing.T) handlerdeps {
+	return fxutil.Test[handlerdeps](
+		t,
+		logimpl.MockModule(),
+		config.MockModule(),
+		fx.Supply(workloadmeta.NewParams()),
+		fx.Supply(context.Background()),
+		workloadmeta.MockModule(),
+		hostnameinterface.MockModule(),
+		flareimpl.MockModule(),
+		dogstatsdServer.MockModule(),
+		serverdebugimpl.MockModule(),
+		fx.Provide(func() optional.Option[logsAgent.Component] {
+			return optional.NewNoneOption[logsAgent.Component]()
+		}),
+		hostimpl.MockModule(),
+		inventoryagentimpl.MockModule(),
+		demultiplexerimpl.MockModule(),
+		inventoryhostimpl.MockModule(),
+		secretsimpl.MockModule(),
+		fx.Provide(func(secretMock secrets.Mock) secrets.Component {
+			component := secretMock.(secrets.Component)
+			return component
+		}),
+		inventorychecksimpl.MockModule(),
+		packagesigningimpl.MockModule(),
+		statusimpl.MockModule(),
+		fx.Provide(func() optional.Option[collector.Component] {
+			return optional.NewNoneOption[collector.Component]()
+		}),
+		eventplatformreceiverimpl.MockModule(),
+		fx.Options(
+			fx.Supply(autodiscoveryimpl.MockParams{Scheduler: nil}),
+			autodiscoveryimpl.MockModule(),
+		),
+	)
+}
+
+func setupRoutes(t *testing.T) *mux.Router {
+	deps := getComponentDeps(t)
+	sender := aggregator.NewNoOpSenderManager()
+
+	router := mux.NewRouter()
+	SetupHandlers(
+		router,
+		deps.FlareComp,
+		deps.Server,
+		deps.ServerDebug,
+		deps.Wmeta,
+		deps.LogsAgent,
+		sender,
+		deps.HostMetadata,
+		deps.InvAgent,
+		deps.Demux,
+		deps.InvHost,
+		deps.SecretResolver,
+		deps.InvChecks,
+		deps.PkgSigning,
+		deps.StatusComponent,
+		deps.Collector,
+		deps.EventPlatformReceiver,
+		deps.Ac,
+	)
+
+	return router
+}
+
+func TestSetupHandlers(t *testing.T) {
+	router := setupRoutes(t)
+	ts := httptest.NewServer(router)
+	defer ts.Close()
+
+	res, err := http.Get(ts.URL + "/version")
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
```

Comment:
This is equivalent to the one line `require.NoError(t, err)`

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/23904/commits/22b5419f3bd559f234d94f099237a791045fc653